wtprompt/core.py

```diff
@@ -20,30 +20,43 @@ def __init__(self, **data):
         # Loading using pydantic validators
         super().__init__(**data)
         self._prompts = {}
+        self._prompt_hashes = {}
 
     def add_prompt(self, prompt_name: str, prompt_text: str):
+        """Add a single prompt, with prompt_name and prompt_text.
+
+        This function stores the prompt and its hash inside the calss.
+        """
         if prompt_name in self._prompts:
             warnings.showwarning(f"Prompt {prompt_name} already present.\n"
                                  f"Please check the prompt names!\nAdding nothing.", Warning)
             return
         self._prompts[prompt_name] = prompt_text
+        # Computing and storing hash
+        computed_hash = hashlib.sha256(prompt_text.encode("utf-8")).hexdigest()
+        self._prompt_hashes[prompt_name] = computed_hash
 
     def _get_prompt_text(self, prompt_name: str):
         return self._prompts[prompt_name]
 
-    def get_prompts(self):
-        return self._prompts
+    def get_prompt(self, prompt_name: str, get_hash: bool=False) -> Union[str, Tuple[str, str]]:
```

Comment:
IF you want to be fancy with typing you can use `typing.overload` to specify that if `get_hash` is `True` then this always returns a `tuple`, if `get_hash` is `False` it always return a `str`.

Alternatively you could have a separate `get_prompt_with_hash` function instead of the `get_hash` flag, which might be better and simpler:

```
def get_prompt_with_hash(self, prompt_name: str) -> Tuple[str, str]:
    return self._prompts[prompt_name], self._prompt_hashes[prompt_name]

def get_prompt(self, prompt_name: str) -> str:
    return self.get_prompt_with_hash(prompt_name)[0]
```